lib/vehicle.rb

```diff
@@ -5,14 +5,20 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :county,
+              :plate_type,
+              :registration_date
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @county = vehicle_details[:county]
+    @plate_type = nil
+    registration_date = nil
```

Comment:
It's typically more conventional to set this to an instance variable here (@registration_date) rather than in your last method